autosubmit/history/database_managers/experiment_history_db_manager.py

```diff
@@ -830,14 +856,41 @@ def get_job_data_max_counter(self, job_name: str = None):
         max_counter = result.maxcounter
         return max_counter if max_counter else DEFAULT_MAX_COUNTER
 
+    def get_jobs_data_last_row(self, job_names) -> dict[str, Any]:
+        job_data_table = get_table_with_schema(self.schema, JobDataTable)
+        jobs_data = self.select_jobs_data(job_data_table, job_names)
+        jobs_data = [dict(job) for job in jobs_data]
+        jobs_data_by_name = {}
+        counters = {}
+        for job in jobs_data:
+            if job['job_name'] not in counters:
+                counters[job['job_name']] = job['counter']
+                jobs_data_by_name[job['job_name']] = job
+            elif job['counter'] > counters[job['job_name']]:
+                counters[job['job_name']] = job['counter']
+                jobs_data_by_name[job['job_name']] = job
+        return jobs_data_by_name
+
+    def select_jobs_data(self, table, job_names) -> list[tuple[str, Any]]:
+        query = select(table).where(
+            and_(
+                table.c.last == 1,
+                table.c.job_name.in_(job_names)
+            )
+        )
+        with self.engine.connect() as conn:
+            rows = conn.execute(query).fetchall()
+        columns = table.c.keys()
+        return [tuple(zip(columns, row)) for row in rows]
```

Comment:
[nitpick] Couldn't it be better to return the result a a dict since it will always be transformed to one in `get_jobs_data_last_row`?

This suggestion should do the job:

```suggestion
        return [row._asdict() for row in rows]
```